src/core/thread/mle.cpp

```diff
@@ -1095,6 +1211,15 @@ void Mle::HandleNetifStateChanged(uint32_t aFlags)
 
         mNetif.GetNetworkDataLocal().SendServerDataNotification();
     }
+
+    if ((aFlags & (OT_NET_ROLE | OT_NET_KEY_SEQUENCE_COUNTER | OT_IP6_ML_ADDR_CHANGED | OT_THREAD_CHILD_ADDED |
+                   OT_THREAD_CHILD_REMOVED)) != 0)
```

Comment:
Rather than storing the entire data every time something changes, I think it is better to only update each item as they change. For example, when a child is added, only add a child entry. When a child is removed, only remove the child entry.